src/Grpc.AspNetCore.Server/GrpcServiceOptions.cs

```diff
@@ -73,6 +76,34 @@ public List<ICompressionProvider> CompressionProviders
             }
             set => _compressionProviders = value;
         }
+
+        /// <summary>
+        /// List of interceptors to be executed. Interceptors executed in order.
+        /// </summary>
+        internal List<Type> Interceptors { get; private set; } = new List<Type>();
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        public void AddInterceptor<T>()
```

Comment:
I think of this as like UseMiddleware, and like UseMiddleware I think we should take a collection of args that get passed to the constructor.

https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.builder.usemiddlewareextensions.usemiddleware